collectors/python.d.plugin/freeswitch/freeswitch.chart.py

```diff
@@ -0,0 +1,150 @@
+            ['DOWN','DOWN','absolute'],
+        ]
+    }
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+
+        self.order = ORDER
+        self.definitions = CHARTS
+
+        self.host = configuration.get('host', DEFAULT_HOST)
+        self.port = configuration.get('port', DEFAULT_PORT)
+        self.password = configuration.get('password', DEFAULT_PASSWORD)
+
+        self.fs = ESL.ESLconnection(self.host, self.port, self.password)
+
+    def _check(self):
+        if not HAS_ESL:
+            self.error("'ESL' package is needed to use freeswitch module, you can install with with pip install python-ESL")
+            return False
+        
+        try:
+            self.fs = ESL.ESLconnection(self.host, self.port, self.password)
+        except:
+            self.error("Cannot connect to FreeSwitch with provided data, check that the service is running")
+            return False
+        
+        return True
+
+    def _get_data(self):
+        if not self.fs.connected():
+            self.fs = ESL.ESLconnection(self.host, self.port, self.password)
+
+        try:
+            data = dict()
+
+            calls = self.fs.api('show calls as json').getBody()
+            registrations = self.fs.api('show registrations as json').getBody()
+            gateways_fs = self.fs.api('sofia status gateway').getBody().splitlines()
+            profiles_fs = self.fs.api('sofia status').getBody().splitlines()
+
+            gateways = {'UNREGED':0,
+                        'TRYING':0,
+                        'REGISTER':0,
+                        'REGED':0,
+                        'UNREGISTER':0,
+                        'FAILED':0,
+                        'FAIL_WAIT':0,
+                        'EXPIRED':0,
+                        'NOREG':0,
+                        'TIMEOUT':0}
+            
+            profiles = {'RUNNING':0,
+                        'DOWN':0}
+            
+            for gateway in gateways_fs:
+                for gw_data in gateway.split():                    
+                    try:
+                        gateways[gw_data]+=1;
+                    except:
+                        continue
+            
+            for profile in profiles_fs:
+                for prof_data in profile.split():                    
+                    try:
+                        profiles[prof_data]+=1;
+                    except:
+                        continue
+
+
+            data['calls']=int(json.loads(calls)["row_count"])
+            data['registrations']=int(json.loads(registrations)["row_count"])
+            for gw_state in gateways:
+                data[gw_state]=int(gateways[gw_state])
+            
+            for prof_state in profiles:
+                data[prof_state]=int(profiles[prof_state])
+
+            return data or None
+        except (ValueError, AttributeError):
```

Comment:
I suggest to simplify it, in general it is not good to have a big block in `try/except`, we need to place there as less code as possible.

> calls = self.fs.api('show calls as json').getBody()
> registrations = self.fs.api('show registrations as json').getBody()
> gateways_fs = self.fs.api('sofia status gateway').getBody().splitlines()
> profiles_fs = self.fs.api('sofia status').getBody().splitlines()

Lets do 4 methods like `get_calls`, `get_regisrations`, etc.